packages/main/src/TabContainer.ts

```diff
@@ -359,10 +149,236 @@ const metadata = {
  * @tagname ui5-tabcontainer
  * @public
  */
+@customElement("ui5-tabcontainer")
+@languageAware
+@fastNavigation
+/**
+ * Fired when a tab is selected.
+ *
+ * @event sap.ui.webc.main.TabContainer#tab-select
+ * @param {HTMLElement} tab The selected <code>tab</code>.
+ * @param {Integer} tabIndex The selected <code>tab</code> index in the flattened array of all tabs and their subTabs, provided by the <code>allItems</code> getter.
+ * @public
+ * @allowPreventDefault
+ */
+@event("tab-select", {
+	detail: {
+		tab: { type: HTMLElement },
+		tabIndex: { type: Number },
+	},
+})
 class TabContainer extends UI5Element {
-	static get metadata() {
-		return metadata;
-	}
+	/**
+	 * Defines whether the tabs are in a fixed state that is not
+	 * expandable/collapsible by user interaction.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TabContainer.prototype.fixed
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	fixed!: boolean;
+
+	/**
+	 * Defines whether the tab content is collapsed.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TabContainer.prototype.collapsed
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	collapsed!: boolean;
+
+	/**
+	 * Defines whether the overflow select list is displayed.
+	 * <br><br>
+	 * The overflow select list represents a list, where all tabs are displayed
+	 * so that it's easier for the user to select a specific tab.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TabContainer.prototype.showOverflow
+	 * @defaultvalue false
+	 * @public
+	 * @deprecated Since the introduction of TabsOverflowMode, overflows will always be visible if there is not enough space for all tabs,
+	 * all hidden tabs are moved to a select list in the respective overflows and are accessible via the <code>overflowButton</code> and / or <code>startOverflowButton</code> slots.
+	 */
+	@property({ type: Boolean })
+	showOverflow!: boolean;
+
+	/**
+	 * Defines the alignment of the content and the <code>additionalText</code> of a tab.
+	 *
+	 * <br><br>
+	 * <b>Note:</b>
+	 * The content and the <code>additionalText</code> would be displayed vertically by default,
+	 * but when set to <code>Inline</code>, they would be displayed horizontally.
+	 *
+	 * <br><br>
+	 * Available options are:
+	 * <ul>
+	 * <li><code>Standard</code></li>
+	 * <li><code>Inline</code></li>
+	 * </ul>
+	 *
+	 * @type {sap.ui.webc.main.types.TabLayout}
+	 * @name sap.ui.webc.main.TabContainer.prototype.tabLayout
+	 * @defaultvalue "Standard"
+	 * @public
+	 */
+	@property({ defaultValue: TabLayout.Standard })
```

Comment:
Could that be overlooked, previously the tabLayout prop used to be defined as `type: String` and it's skipped here as expected, but I think it should be type: TabLayout in the first place. Moreover, the jsdoc suggests the same ("@type {sap.ui.webc.main.types.TabLayout}").